src/Interpreters/executeQuery.cpp

```diff
@@ -355,6 +355,15 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
 {
     const auto current_time = std::chrono::system_clock::now();
 
+    auto & client_info = context->getClientInfo();
+
+    // If it's an initial query, set to current_time
+    if (!internal && client_info.initial_query_start_time == 0)
```

Comment:
Checking `QueryKind::INITIAL_QUERY` might be more natural.

Reply:
I'll add proper comments to explain it.